Tests/NIOTests/DatagramChannelTests.swift

```diff
@@ -667,8 +667,13 @@ final class DatagramChannelTests: XCTestCase {
         } ())
     }
     
-    private func testEcnReceive(address: String, vectorRead: Bool, vectorSend: Bool) {
+    private func testEcnReceive(address: String, vectorRead: Bool, vectorSend: Bool, receivePacketInfo: Bool = false) {
         XCTAssertNoThrow(try {
+            // Fake sending packet to self on the loopback interface if requested
+            let fakeDestinationAddress = try SocketAddress(ipAddress: address, port: 0)
+            let fakeIngressIfaceIndex = 1
```

Comment:
I think this risks breaking in CI: as the write path doesn't do anything, and as these are real channels, we actually need to know the real interface index. You can get this out of [`System.enumerateDevices`](https://apple.github.io/swift-nio/docs/current/NIO/Enums/System.html#/s:3NIO6SystemO16enumerateDevicesSayAA16NIONetworkDeviceVGyKFZ), which will let you search for the right one. A simple `filter` should find the interface we're using.